src/pilotdatasync-xp11.cpp

```diff
@@ -102,6 +108,64 @@ void dummy_key_handler(
     int losing_focus
 ) {}
 
+int runTCPThread(){
+    TCPClient client; 
+    ThreadQueue myQueue; 
+    
+    std::cout << "TCP Client Starting..." << std::endl;
+    bool stop_exec = false; 
+
+    while(stop_exec == false){
+        if(myQueue.size() == 0){
+            std::this_thread::sleep_for(150ms);
+            continue;
+        }
+
+        ThreadMessage tm = myQueue.pop();
+        if(tm.end_execution_flag == true){
+            stop_exec == false;
+        } else {
+            vector <string> myVec; 
+            for (int i = 0; i < 4; i++){
+                myVec.push_back(to_string(tm.values_for_packet[i]));
+            }
+            string packet = generate_packet(myVec);
+
+            //move along the TCP 
+            client.sendData(packet);
+        }
+    }
+
+    if (!client.initializeWinsock()) {
+        std::cout << "Failed to initialize Winsock" << std::endl;
+        return 1; 
+    }
+    
+    if (!client.createSocket()) {
+        std::cout << "Failed to create socket" << std::endl;
+        return 1; 
+        }
+        
+
+    std::string serverIP = "127.0.0.1";  // Changed to IPv4 localhost
+    if (!client.connectToServer(serverIP)) {
+        std::cout << "Failed to connect to server" << std::endl;
+        return 1;
+    }
+    
+    // Send test messages
```

Comment:
Can we get this switched to sending test packets using the `generate_packet` function I wrote in `threadig-tools.cpp`?